docs/source/common_issues.rst

```diff
@@ -808,3 +808,78 @@ not necessary:
     class NarrowerArgument(A):
         def test(self, t: List[int]) -> Sequence[str]:  # type: ignore[override]
             ...
+
+TypeVars are not the same
+------------------------------------------------
+
+As discussed in the issue
+`TypeVars should be allowed to be the same <https://github.com/python/mypy/issues/7864>`_,
+TypeVars are *not* equivalent even if they are identically defined.
+
+Here is an example using two TypeVars that are unconstrained and unbound:
+
+.. code-block:: python
+
+    from typing import Callable, Iterable, TypeVar
+
+    A = TypeVar("A")
+    B = TypeVar("B")
+
+    def identity(x: A) -> A:
+        return x
+
+    # ok
+    def _map(queue: Iterable[B], function: Callable[[B], A]) -> Iterable[A]:
+        return map(function, queue)
+
+    # fails
+    def _map2(queue: Iterable[B], function: Callable[[B], A] = identity) -> Iterable[A]:
+        return map(function, queue)
+
+
+The signature of ``_map2()`` triggers
+``error: Incompatible default for argument "function" (default has type "Callable[[A], A]",``
+``argument has type "Callable[[B], A]")``
+
+Likewise, this error will occur for constrained and bound TypeVars as well.
+
+.. code-block:: python
+
+    A = TypeVar("A", int, str)
+    B = TypeVar("B", int, str)
+    # or
+    A = TypeVar("A", bound=complex)
+    B = TypeVar("B", bound=complex)
+
+
+In some cases, the ``@overload`` decorator can be used to provide the desired type checking. Here's an example.
```

Comment:
Style nit: write "... an example:" (with a colon)

A simpler workaround would be to cast the default value to the desired type using `cast`.

Reply:
I was able to eliminate the second overload def:

`def map_2(queue: Iterable[A], function: Callable[[A], A]) -> Iterable[A]: ...`

by using a cast like so:

`map_2(list_2, cast(Callable[[B], A], identity))`

Is this what you meant?

If so, I would prefer the original approach of keeping the extra overload def so that the actual call is easier to read and to understand.

Reply:
Hi, I'm not sure whether this is the workaround that @JukkaL suggested. 
```
def map2(queue: Iterable[B], function: Callable[[B], A] = cast(Callable[[B], A], identity)) -> Iterable[A]: ...
```